lib/tasks/link_checker.rake

```diff
@@ -1,28 +1,29 @@
 require "uri"
 
 def gather_urls object
-  urls = []
+  urls = Set[]
   if object.class == String
-    urls.push(*URI.extract(object, %w[http https]))
+    urls.merge(Set.new(URI.extract(object, %w[http https]).to_set))
   elsif object.class == Hash
     object.each do |_, value|
-      urls.push(*gather_urls(value))
+      urls.merge(gather_urls(value))
     end
   elsif object.class == Array
     object.each do |value|
-      urls.push(*gather_urls(value))
+      urls.merge(gather_urls(value))
     end
   end
   urls
 end
 
 def get_urls_from_locale_files
-  urls = Dir.glob("config/locales/*.yml").map { |locale_filename| gather_urls YAML.safe_load(File.read(locale_filename)) }
-  urls.flatten!.uniq!
+  urls = Dir.glob("config/locales/*.yml")
+    .map { |locale_filename| gather_urls YAML.safe_load(File.read(locale_filename)) }
+    .reduce(:+)
```

Comment:
What does `:+` do?

Reply:
Are you trying to flatten the array? If so, then `flat_map` would work just as well, and be easier to read.